modules/extension/wms/src/main/java/org/geotools/ows/wms/map/WMSCoverageReader.java

```diff
@@ -207,6 +212,31 @@ void addLayer(Layer layer, String style) {
         updateBounds();
     }
 
+    /** Returns the first CRS from layer srs list that matches with bounding boxes map keys. */
+    private String getPriorityCRS(Layer layer) {
+        List<String> priorityCRSs = getPriorityCRSs(layer);
+        if (!priorityCRSs.isEmpty()) {
+            String srs = priorityCRSs.get(0);
+            try {
+                // check it's valid, if not we crap out and move to the next
```

Comment:
"move to the next" but there is no loop. Next what? Next layer?

Reply:
Fixed, indeed we need to iterate CRSs with BBOXs in case the first ones result in CRS decode exception.